contracts/contracts/validator-registry/VanillaRegistry.sol

```diff
@@ -330,16 +343,18 @@ contract VanillaRegistry is IVanillaRegistry, VanillaRegistryStorage,
             require(_isUnstaking(pubKey), IVanillaRegistry.MustUnstakeToWithdraw());
             require(block.number > validator.unstakeOccurrence.blockHeight + unstakePeriodBlocks,
                 IVanillaRegistry.WithdrawingTooSoon());
+            require(validator.withdrawalAddress == expectedWithdrawalAddress,
+                IVanillaRegistry.WithdrawalAddressMismatch(validator.withdrawalAddress, expectedWithdrawalAddress));
             uint256 balance = validator.balance;
-            require(balance != 0, IVanillaRegistry.NothingToWithdraw());
             totalAmount += balance;
             delete stakedValidators[pubKey];
-            // msg.sender must be withdrawal address from wrapping onlyWithdrawalAddress modifier
-            emit StakeWithdrawn(msg.sender, pubKey, balance);
+            emit StakeWithdrawn(msg.sender, expectedWithdrawalAddress, pubKey, balance);
+        }
+        if (totalAmount != 0) {
+            (bool success, ) = expectedWithdrawalAddress.call{value: totalAmount}("");
```

Comment:
Transfer of funds always goes to withdrawal address, even if `msg.sender` is the owner account